### PR TITLE
[env](compile)open compile check in columns class

### PR DESCRIPTION
**Pull request by xinyiZzz**
### What problem does this PR solve?

Problem Summary:

open compile check in columns class

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [x] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/vec/common/pod_array.h [hunk 1]
```diff
@@ -43,6 +43,7 @@
 #include "vec/common/pod_array_fwd.h"
 
 namespace doris::vectorized {
+// #include "common/compile_check_begin.h"
```
Comment (HappenLee) [inline be/src/vec/common/pod_array.h:46]: just remove

Comment (xinyiZzz) [reply be/src/vec/common/pod_array.h]: done

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/vec/columns/column_string.cpp [hunk 2]
```diff
@@ -391,8 +402,11 @@ ColumnPtr ColumnStr<T>::permute(const IColumn::Permutation& perm, size_t limit)
 template <typename T>
 StringRef ColumnStr<T>::serialize_value_into_arena(size_t n, Arena& arena,
                                                    char const*& begin) const {
-    uint32_t string_size(size_at(n));
-    uint32_t offset(offset_at(n));
+    // https://github.com/apache/doris/pull/10832
+    // Use uint32 instead of size_t to reduce agg key's length.
+    // Size of a line will not exceed 4G, check the size when inserting data into ColumnStr<T>.
+    auto string_size(cast_set<uint32_t>(size_at(n)));
```
Comment (HappenLee) [inline be/src/vec/columns/column_string.cpp:408]: why here need to check, seem only do check in insert is enough

File: be/src/vec/columns/column_string.cpp [hunk 3]
```diff
@@ -434,8 +448,11 @@ template <typename T>
 void ColumnStr<T>::serialize_vec(std::vector<StringRef>& keys, size_t num_rows,
                                  size_t max_row_byte_size) const {
     for (size_t i = 0; i < num_rows; ++i) {
-        uint32_t offset(offset_at(i));
-        uint32_t string_size(size_at(i));
+        // https://github.com/apache/doris/pull/10832
```
Comment (HappenLee) [inline be/src/vec/columns/column_string.cpp:451]: same to the up

File: be/src/vec/columns/column_vector.h [hunk 4]
```diff
@@ -178,9 +180,11 @@ class ColumnVector final : public COWHelper<IColumn, ColumnVector<T>> {
     void insert_range_of_integer(T begin, T end) {
         if constexpr (std::is_integral_v<T>) {
             auto old_size = data.size();
-            data.resize(old_size + (end - begin));
-            for (int i = 0; i < end - begin; i++) {
-                data[old_size + i] = begin + i;
+            // (end - begin) is the number of elements added to the vector,
```
Comment (HappenLee) [inline be/src/vec/columns/column_vector.h:183]: better use `std::iota`

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/vec/columns/column_vector.h (hunk 4 above)
Comment (xinyiZzz) [reply be/src/vec/columns/column_vector.h]: done

File: be/src/vec/columns/column_string.cpp (hunk 3 above)
Comment (xinyiZzz) [reply be/src/vec/columns/column_string.cpp]: done

File: be/src/vec/columns/column_string.cpp (hunk 2 above)
Comment (xinyiZzz) [reply be/src/vec/columns/column_string.cpp]: done

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Mryange — APPROVED** [review]

**Pull request merged**